go-controller/pkg/node/healthcheck.go

```diff
@@ -93,29 +96,120 @@ func countLocalEndpoints(ep *kapi.Endpoints, nodeName string) int {
+	stdout, _, err := util.RunOVSVsctl("--data=bare", "--no-headings", "--columns=name", "find",
+		"interface", "ofport=-1")
+	if err != nil {
+		klog.Errorf("Failed to list OVS interfaces with ofport set to -1")
+		return
+	}
+	if len(stdout) == 0 {
+		return
+	}
+
+	values := strings.Split(stdout, "\n\n")
+	for _, val := range values {
+		klog.Warningf("Found stale interface %s, so deleting it", val)
+		_, stderr, err := util.RunOVSVsctl("--if-exists", "--with-iface", "del-port", val)
+		if err != nil {
+			klog.Errorf("Failed to delete OVS port/interface %s: stderr: %s (%v)",
+				val, stderr, err)
+		}
+	}
+}
+
+// checkForStaleOVSRepresentorInterfaces checks for stale OVS ports backed by Repreresentor interfaces,
+// derrive iface-id from pod name and namespace then remove any interfaces assoicated with a sandbox that are
+// not scheduled to the node.
+func checkForStaleOVSRepresentorInterfaces(nodeName string, k kube.Interface) {
+	// list Ports on br-int and find out the current ports that have iface-ids
+	currentIfaceIds := make(map[string]string)
+	ports, stderr, err := util.RunOVSVsctl("list-ports", "br-int")
+	if err != nil {
+		klog.Errorf("failed to get list of ports on br-int:, stderr: %q, error: %v", stderr, err)
+		return
+	}
+
+	for _, port := range strings.Split(ports, "\n") {
+		stdout, stderr, err := util.RunOVSVsctl("get", "Port", port, "Interfaces")
+		if err != nil {
+			klog.Errorf("failed to get port %q on br-int:, stderr: %q, error: %v", port, stderr, err)
+			continue
+		}
+		// remove brackets on list of interfaces
+		ifaces := strings.TrimPrefix(strings.TrimSuffix(stdout, "]"), "[")
+		for _, iface := range strings.Split(ifaces, ",") {
+			// check that interface is associated with a sandbox
+			sandbox, stderr, err := util.RunOVSVsctl("get", "Interface", strings.TrimSpace(iface),
+				"external_ids:sandbox")
 			if err != nil {
-				klog.Errorf("Failed to list OVS interfaces with ofport set to -1")
+				if !strings.Contains(stderr, "no key") {
+					klog.Errorf("failed to get Interface %q external_ids:sandbox. stderr: %q, error: %v",
+						iface, stderr, err)
+				}
 				continue
 			}
-			if len(stdout) == 0 {
+			if sandbox == "" {
 				continue
 			}
-			values := strings.Split(stdout, "\n\n")
-			for _, val := range values {
-				klog.Warningf("Found stale interface %s, so deleting it", val)
-				_, stderr, err := util.RunOVSVsctl("--if-exists", "--with-iface", "del-port", val)
-				if err != nil {
-					klog.Errorf("Failed to delete OVS port/interface %s: stderr: %s (%v)",
-						val, stderr, err)
-				}
+			// get iface ID
+			ifaceId, stderr, err := util.RunOVSVsctl("get", "Interface", strings.TrimSpace(iface),
+				"external_ids:iface-id")
+			if err != nil {
+				klog.Errorf("failed to get Interface %q external_ids:iface-id. stderr: %q, error: %v",
+					iface, stderr, err)
+				continue
+			}
+			if ifaceId == "" {
+				continue
+			}
+			currentIfaceIds[ifaceId] = port
+		}
+	}
```

Comment:
@adrianchiris  You can simplify L135-170 with:
```
 ovs-vsctl --columns name,external_ids --data=bare --no-headings --format=csv find interface 'external_ids:sandbox!=""'
```
That will return you all the interfaces  with `sandbox` key set, so essentially k8s pods. Next, you parse the output to get the`iface-id` value.

You can then delete the stale OVS Port with
`ovs-vsctl --with-iface del-port  <ifaceName>`

So, you don't need to get the Port Name at all.

Reply:
Furthermore, imagine a 1000-Node k8s cluster. With your approach, we will always list 1000 Geneve Ports from `ovs-vsctl list-ports br-int` and then discard it since `sandbox` will not be set. So, it is better to filter the results on the OVSDB server side.

Reply:
better to do the filtering ovsdb side it probably does it faster and more efficient than this.